src/components/ImportForm/GitImportForm.tsx

```diff
@@ -67,11 +76,36 @@ const GitImportForm: React.FunctionComponent<React.PropsWithChildren<GitImportFo
     },
   };
 
+  React.useEffect(() => {
+    const callback = (e) => {
+      setSubmitModalText(e.detail.text);
+      setValueProgressBar(e.detail.progressValue);
+    };
+    document.addEventListener(IMPORT_EVENT_NAME, callback);
+    return () => {
+      document.removeEventListener(IMPORT_EVENT_NAME, callback);
+    };
+  }, []);
+
   const showModal = useModalLauncher();
 
   const handleSubmit = React.useCallback(
     (values: ImportFormValues, formikHelpers: any) => {
       track(TrackEvents.ButtonClicked, { link_name: 'import-submit', workspace });
+      const detectedComponents = values.selectedComponents
+        ? values.components?.filter((value, index) => values.selectedComponents[index])
+        : values.components;
+      if (detectedComponents.length > 5) {
```

Comment:
I think it would be better if we show the modal all the time and be consistent with the experience.